flocker/provision/_install.py

```diff
@@ -517,48 +516,48 @@ def task_open_control_firewall(distribution):
 
 
 def task_enable_flocker_agent(distribution, control_node,
-                              dataset_backend=DatasetBackend.zfs,
-                              dataset_backend_configuration=dict(
-                                  pool=u'flocker'
-                              )):
+                              dataset_backend=None,
+                              dataset_backend_configuration=None):
     """
     Configure and enable the flocker agents.
 
     :param bytes control_node: The address of the control agent.
     :param DatasetBackend dataset_backend: The volume backend the nodes are
-        configured with.  (This has a default for use in the documentation).
+        configured with.
     :param dict dataset_backend_configuration: The backend specific
         configuration options.
     """
-    dataset_backend_configuration = dataset_backend_configuration.copy()
-    dataset_backend_configuration.update({
-        u"backend": dataset_backend.name,
-    })
-
-    put_config_file = put(
-        path='/etc/flocker/agent.yml',
-        content=yaml.safe_dump(
-            {
-                "version": 1,
-                "control-service": {
-                    "hostname": control_node,
-                    "port": 4524,
+    initial_sequence = []
+    if dataset_backend is not None:
+        dataset_backend_configuration = dataset_backend_configuration.copy()
+        dataset_backend_configuration.update({
+            u"backend": dataset_backend.name,
+        })
+
+        put_config_file = put(
+            path='/etc/flocker/agent.yml',
+            content=yaml.safe_dump(
+                {
+                    "version": 1,
+                    "control-service": {
+                        "hostname": control_node,
+                        "port": 4524,
+                    },
+                    "dataset": dataset_backend_configuration,
                 },
-                "dataset": dataset_backend_configuration,
-            },
-        ),
-    )
+            ),
+        )
+        initial_sequence.append(put_config_file)
```

Comment:
`intial_sequence` should be renamed to something descriptive. `put_config_file` or similar perhaps.